src/integration/java/uk/nhs/prm/repo/ehrtransferservice/database/TransferServiceTest.java

```diff
@@ -113,22 +163,27 @@ void updateConversationTransferStatus_ValidInboundConversationIdAndConversationT
         final RepoIncomingEvent event = createRepoIncomingEvent(inboundConversationId);
 
         // when
-        transferService.createConversation(event);
+        try {
+            transferService.createConversationOrResetForRetry(event);
+        } catch (ConversationIneligibleForRetryException e) {
+            fail("Conversation should be new and eligible.");
+        }
+
         transferService.updateConversationTransferStatus(inboundConversationId, INBOUND_FAILED);
         ConversationRecord record = transferService
             .getConversationByInboundConversationId(inboundConversationId);
 
         // then
-        assertEquals(record.state(), INBOUND_FAILED.name());
+        assertEquals(record.transferStatus(), INBOUND_FAILED);
```

Comment:
Should we have an additional step in here?
```
        try {
            transferService.createConversationOrResetForRetry(event);
            fail("Conversation should not be new and eligible.");
        } catch (ConversationIneligibleForRetryException e) {
            // Expected
        }
```

Reply:
Created new test for this case: updateConversationTransferStatus_ConversationIsAlreadyComplete_ShouldThrowExceptionAndNotUpdateTransferStatus